File: KadirliApp.Domain/Entities/Pharmacy.cs
using KadirliApp.Domain.Common;

namespace KadirliApp.Domain.Entities;

public class Pharmacy : BaseEntity
{
    public string Name { get; set; } = default!;
    public string? Address { get; set; }
    public string? Phone { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    public string? WorkingHours { get; set; }
    public string? PharmacistName { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<PharmacySchedule> Schedules { get; set; } = new List<PharmacySchedule>();
}

File: mobile/lib/features/pharmacies/data/models/pharmacy.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'pharmacy.freezed.dart';
part 'pharmacy.g.dart';

/// `GET /v1/pharmacies` ve `/v1/pharmacies/{id}` gövdesi (`PharmacyDto`).
///
/// Liste ve detay **aynı şekli** döndürüyor (canlıda doğrulandı) — ayrı bir
/// detay modeli tutmaya gerek yok.
@freezed
abstract class Pharmacy with _$Pharmacy {
  const factory Pharmacy({
    required String id,
    required String name,
    String? address,
    String? phone,
    double? latitude,
    double? longitude,

    /// "08:30 - 19:00" gibi serbest metin (saat aritmetiği yapma).
    String? workingHours,
    String? pharmacistName,
    @Default(true) bool isActive,
  }) = _Pharmacy;

  const Pharmacy._();

  factory Pharmacy.fromJson(Map<String, dynamic> json) =>
      _$PharmacyFromJson(json);

  bool get hasLocation => latitude != null && longitude != null;

  bool get hasPhone => (phone ?? '').trim().isNotEmpty;
}
